src/singleSpaGenerator/singleSpaGenerator.ts

```diff
@@ -0,0 +1,83 @@
+import FullEnvironment from "yeoman-environment";
+import { log, secondary } from "../cli-logger";
+import p from "node:path";
+
+const env = new FullEnvironment();
+export const runSingleSpaGenerator = async (args: CreateSingleSpaArgument) => {
+  let arg = " ";
+  arg += `--dir ${args.dir} `;
+  arg += `--framework ${args.framework} `;
+  arg += `--orgName ${args.orgName} `;
+  arg += `--projectName ${args.projectName} `;
+  arg += `--moduleType ${args.moduleType} `;
+  arg += `--packageManager ${args.packageManager} `;
+
+  if (args.framework == "react") {
+    arg += `--typescript ${args.typescript} `;
+  }
+
+  log(secondary.inverse("RUNNING") + secondary(` create-single-spa ${arg}`));
+
+  const generator = await import("generator-single-spa");
+  // @ts-ignore
+  env.registerStub(generator.default, "generator-single-spa");
+  // @ts-ignore
+  await env.run(`generator-single-spa ${arg} --skipInstall`, "");
```

Comment:
I believe `generator-single-spa` is available via an npm package - I'd prefer running it within the same os process if possible, by calling it as a JS api. If we need to PR create-single-spa to export the correct functions, let's do so.

Reply:
baseplate cli is typescript and `generator-single-spa` is javascript which makes it harder to rely on the API, I actually started to doing it that way, I'm not sure if using `yeoman-environment` spawn child process , but doing it at the generator level has some benefits such as not being dependent on js api of `generator-single-spa` instead working with it's console interface. the user/dev experience is seamless as it will prompt for any missing args and baseplate cli resumes once generator is done. I think we may need to discuss this in more details.